Sources/OfflineEntitlements/CustomerInfo+OfflineEntitlements.swift

```diff
@@ -0,0 +1,68 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  CustomerInfo+OfflineEntitlements.swift
+//
+//  Created by Nacho Soto on 3/21/23.
+
+import Foundation
+
+@available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
+extension CustomerInfo {
+
+    convenience init(
+        from purchasedSK2Products: [PurchasedSK2Product],
+        mapping: ProductEntitlementMapping,
+        sandboxEnvironmentDetector: SandboxEnvironmentDetector = BundleSandboxEnvironmentDetector.default
+    ) {
+        let subscriber = CustomerInfoResponse.Subscriber(
+            originalAppUserId: IdentityManager.generateRandomID(),
```

Comment:
should this be the current app user id?

Reply:
This isn't known at this point (this part of the code doesn't depend on any existing `CustomerInfo`).
I think that's something we can do when we merge with the existing one.

Reply:
But there would always be a non nil app user id, either the one configured or the one logged in. We can use that one without checking any other CustomerInfo

Reply:
I guess I'll inject the ID here for now.

Reply:
Done.